Split/Network/Streaming/RetryableSplitsUpdateWorkerFactory.swift

```diff
@@ -104,35 +104,38 @@ class DefaultSyncWorkerFactory: SyncWorkerFactory {
             timer: DefaultPeriodicTimer(interval: splitConfig.segmentsRefreshRate), eventsManager: eventsManager)
     }
 
-    func createPeriodicImpressionsRecorderWorker(syncHelper: ImpressionsRecorderSyncHelper) -> PeriodicRecorderWorker {
+    func createPeriodicImpressionsRecorderWorker(syncHelper: ImpressionsRecorderSyncHelper?) -> PeriodicRecorderWorker {
         let impressionWorker = ImpressionsRecorderWorker(impressionsStorage: storageContainer.impressionsStorage,
                                                          impressionsRecorder: apiFacade.impressionsRecorder,
-                                                         impressionsPerPush: Int(splitConfig.impressionsChunkSize))
+                                                         impressionsPerPush: Int(splitConfig.impressionsChunkSize),
+                                                         impressionsSyncHelper: syncHelper)
 
         let timer = DefaultPeriodicTimer(deadline: 0, interval: splitConfig.impressionRefreshRate)
         return DefaultPeriodicRecorderWorker(timer: timer, recorderWorker: impressionWorker)
     }
 
-    func createImpressionsRecorderWorker(syncHelper: ImpressionsRecorderSyncHelper) -> RecorderWorker {
+    func createImpressionsRecorderWorker(syncHelper: ImpressionsRecorderSyncHelper?) -> RecorderWorker {
         return ImpressionsRecorderWorker(impressionsStorage: storageContainer.impressionsStorage,
                                                          impressionsRecorder: apiFacade.impressionsRecorder,
                                                          impressionsPerPush: Int(splitConfig.impressionsChunkSize))
```

Comment:
javi, check if we should also pass `impressionsSyncHelper` here (I see we are doing it for `eventsSyncHelper` in both `createPeriodicEventsRecorderWorker` and `createEventsRecorderWorker` )